File: frontend/src/microservices/content/objects/ContentImagePlannerResponse.ts
// ContentImagePlannerResponse：content 域评论图片上传返回对象。

export type ContentImagePlannerResponse = {
  imageId: string
  publicUrl: string
  originalFileName: string
  sortOrder: number
  createdAt: string
}

export const contentImagePlannerResponseFromJson = (json: string): ContentImagePlannerResponse =>
  JSON.parse(json) as ContentImagePlannerResponse

export const contentImagePlannerResponseToJson = (value: ContentImagePlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/ContentImagePlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ContentImagePlannerResponse(
    imageId: String,
    publicUrl: String,
    originalFileName: String,
    sortOrder: Int,
    createdAt: String
)

object ContentImagePlannerResponse:
  given sourceEncoder: Encoder[ContentImagePlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ContentImagePlannerResponse] = deriveDecoder
